WordPress/openverse
# Add source links to creator pages

## Fixes



Fixes #3496 by @AetherUnbound 

## Description


This PR updates the collection header component to match the updated designs.

The creator page has a scrollable area with a button linking to the source page in Openverse.

On narrow screens, the icons next to the `h1` heading are replaced with the words. To make this heading accessible, I wrapped the `h1` around the icon/collection name, as well as the name of the tag/source/creator. This is not perfect, especially for i18n, but it gives the screen reader user a better context.

The newly-added source button is scrollable to allow for long source names. To implement this, I extracted the scrolling features from `VByLine` into a separate component, and reused it for this button as well. Some of the extra margin/padding values were added to keep the height of the button correct while also making the focus ring visible (this is a problem for buttons in containers that have overflow set to anything other than visible.

~~This PR adds a link to the creator view results label. Due to limitations in how v.8 vue-i18n's `i18n` functional component works, we cannot use it with the pluralization settings (we need pluralization because the label has "x results"). This is why I didn't use `<i18n path="the creator label">` but created a workaround splitting the creator result label by the source name instead.~~

## Testing Instructions

Go to `/preferences` page and make sure the `additional_search_views` switch is `on`.
Go to a single result page, and click on a creator link button underneath the image or audio. On the source view, you should see the link to a source with an "external link" icon next to it. When you click it, you should see an analytics event logged in the console (and should see the source page open in a new tab).

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
